app/utilities/githubApi/index.js

```diff
@@ -8,10 +8,28 @@ import Notifier from '../notifier'
 import { remote } from 'electron'
 const logger = remote.getGlobal('logger')
 
+// TODO: move to global configuration
+let Network = null
+try {
+  Network = require('../../../configs/network')
+} catch (e) {
+  if (e.code !== 'MODULE_NOT_FOUND') throw e
+  Network = require('../../../configs/networkDummy')
+}
+
+import ProxyAgent from 'proxy-agent'
+let proxyAgent = null
+const proxyUri = process.env.http_proxy || Network.proxy
+if (proxyUri) {
+  logger.info("use proxy", proxyUri)
+  proxyAgent = new ProxyAgent(proxyUri)
+}
+
 function exchangeAccessToken (clientId, clientSecret, authCode) {
   return ReqPromise({
     method: 'POST',
     uri: 'https://github.com/login/oauth/access_token',
+    agent: proxyAgent,
```

Comment:
I tested, if `proxyAgent` is `null`, requests will use system default proxy settings.
https://github.com/request/request/blob/master/request.js#L462
It uses (!agent) like statement to detect the option, `null` or `false` ought to be both working?